sw/device/lib/dif/dif_mbx.c

```diff
@@ -0,0 +1,130 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_mbx.h"
+
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+
+#include "mbx_regs.h"  // Generated.
+
+dif_result_t dif_mbx_range_set(const dif_mbx_t *mbx,
+                               dif_mbx_range_config_t config) {
+  if (mbx == NULL) {
+    return kDifBadArg;
+  }
+  // Note: the limit addresses are _inclusive_, specifying the start address of
+  // the final valid DWORD.
+  if (config.imbx_base_addr > config.imbx_limit_addr) {
+    return kDifBadArg;
+  }
+  if (config.ombx_base_addr > config.ombx_limit_addr) {
+    return kDifBadArg;
+  }
+  // Check that the inbound mailbox and outbound mailbox memory ranges do not
+  // collide with each other.
+  if ((config.imbx_base_addr <= config.ombx_limit_addr) &&
+      (config.ombx_base_addr <= config.imbx_limit_addr)) {
+    return kDifBadArg;
+  }
+
+  mmio_region_write32(mbx->base_addr, MBX_INBOUND_BASE_ADDRESS_REG_OFFSET,
+                      config.imbx_base_addr);
+  mmio_region_write32(mbx->base_addr, MBX_INBOUND_LIMIT_ADDRESS_REG_OFFSET,
+                      config.imbx_limit_addr);
+  mmio_region_write32(mbx->base_addr, MBX_OUTBOUND_BASE_ADDRESS_REG_OFFSET,
+                      config.ombx_base_addr);
+  mmio_region_write32(mbx->base_addr, MBX_OUTBOUND_LIMIT_ADDRESS_REG_OFFSET,
+                      config.ombx_limit_addr);
+  // Indicate the range configuration to be valid.
+  mmio_region_write32(mbx->base_addr, MBX_ADDRESS_RANGE_VALID_REG_OFFSET, 1);
+
+  return kDifOk;
+}
+
+dif_result_t dif_mbx_range_get(const dif_mbx_t *mbx,
+                               dif_mbx_range_config_t *config) {
+  if (mbx == NULL || config == NULL) {
+    return kDifBadArg;
+  }
+
+  config->imbx_base_addr =
+      mmio_region_read32(mbx->base_addr, MBX_INBOUND_BASE_ADDRESS_REG_OFFSET);
+  config->imbx_limit_addr =
+      mmio_region_read32(mbx->base_addr, MBX_INBOUND_LIMIT_ADDRESS_REG_OFFSET);
+  config->ombx_base_addr =
+      mmio_region_read32(mbx->base_addr, MBX_OUTBOUND_BASE_ADDRESS_REG_OFFSET);
+  config->ombx_limit_addr =
+      mmio_region_read32(mbx->base_addr, MBX_OUTBOUND_LIMIT_ADDRESS_REG_OFFSET);
+  return kDifOk;
+}
+
+dif_result_t dif_mbx_is_busy(const dif_mbx_t *mbx, bool *is_busy) {
+  if (mbx == NULL || is_busy == NULL) {
+    return kDifBadArg;
+  }
+
+  *is_busy = bitfield_bit32_read(
+      mmio_region_read32(mbx->base_addr, MBX_STATUS_REG_OFFSET),
+      MBX_STATUS_BUSY_BIT);
```

Comment:
I would avoid the implicit cast from `int` to` bool`.
```suggestion
  *is_busy = (bitfield_bit32_read(
      mmio_region_read32(mbx->base_addr, MBX_STATUS_REG_OFFSET),
      MBX_STATUS_BUSY_BIT) == 1);
```